the OPEN_MAX define has been removed in newer kernel headers so use the
proper method of getting the value dynamically

http://bugs.gentoo.org/195861

--- rdisc.c
+++ rdisc.c
@@ -247,7 +247,7 @@ void do_fork(void)
 	if ((pid=fork()) != 0)
 		exit(0);
 
-	for (t = 0; t < OPEN_MAX; t++)
+	for (t = 0; t < sysconf(_SC_OPEN_MAX); t++)
 		if (t != s)
 			close(t);

